misc/python/materialize/mzbuild.py

```diff
@@ -1009,10 +1011,22 @@ def acquire(self, max_retries: int | None = None) -> None:
 
         # Only retry in CI runs since we struggle with flaky docker pulls there
         if not max_retries:
-            max_retries = 3 if ui.env_is_truthy("CI") else 1
+            max_retries = 5 if ui.env_is_truthy("CI") else 1
         assert max_retries > 0
 
-        deps_to_build = [dep for dep in self if not dep.try_pull(max_retries)]
+        deps_to_build = [
+            dep for dep in self if not dep.publish or dep.try_pull(max_retries)
+        ]
+
+        # Don't attempt to build in CI, as our timeouts and small machines won't allow it anyway
+        if ui.env_is_truthy("CI"):
+            expected_deps = [dep for dep in deps_to_build if dep.publish]
+            if expected_deps:
+                print(
+                    f"+++ Expected builds to be available, the build probably failed, so not proceeding: {expected_deps}"
+                )
+                sys.exit(5)
```

Comment:
Unique return code, so far not used, but in the future we could do auto-retries in mkpipeline.py based on it.